appleauth/auth_source.go

```diff
@@ -0,0 +1,153 @@
+package appleauth
+
+import (
+	"fmt"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-steplib/steps-deploy-to-itunesconnect-deliver/devportalservice"
+)
+
+// Source returns a specific kind (Apple ID/API Key) Apple authentication data from a specific source (Bitrise Service, manual input)
+type Source interface {
+	Fetch(connection *devportalservice.AppleDeveloperConnection, inputs Inputs) (*Credentials, error)
+	Description() string
+	RequiresConnection() bool
+}
+
+// ConnectionAPIKeySource provides API Key from Bitrise Service
+type ConnectionAPIKeySource struct{}
+
+// InputAPIKeySource provides API Key from manual input
+type InputAPIKeySource struct{}
+
+// ConnectionAppleIDSource provides Apple ID from Bitrise Service
+type ConnectionAppleIDSource struct{}
+
+// InputAppleIDSource provides Apple ID from manual input
+type InputAppleIDSource struct{}
+
+// Description ...
+func (*ConnectionAPIKeySource) Description() string {
+	return "Connected Apple Developer Portal Account for App Store Connect API found"
+}
+
+// RequiresConnection ...
+func (*ConnectionAPIKeySource) RequiresConnection() bool {
+	return true
+}
+
+// Fetch ...
+func (*ConnectionAPIKeySource) Fetch(conn *devportalservice.AppleDeveloperConnection, inputs Inputs) (*Credentials, error) {
+	if conn == nil || conn.JWTConnection == nil { // Not configured
+		return nil, nil
+	}
+
+	return &Credentials{
+		APIKey: conn.JWTConnection,
+	}, nil
+}
+
+//
+
+// Description ...
+func (*InputAPIKeySource) Description() string {
+	return "Authenticating using Step inputs (App Store Connect API)"
```

Comment:
`API key authentication-related inputs found`

Reply:
since the API key other than the App Store Connect API, is used for ITMSTransporter